Cities order

kent1D opened the pull request:
Change strategy to order cities

First, we order them with capital field (an admin_level 4 capital should
be shown before bigger cities)

Then we use city / town tag to order them

Then we use the population tag.

Should FIX #36

Additionally: Use shield-min-padding to avoid cutted letters

kent1D committed:
Change strategy to order cities

First, we order them with capital field (an admin_level 4 capital should
be shown before bigger cities)

Then we use city / town tag to order them

Then we use the population tag.

Should FIX #36

kent1D committed:
Use shield-min-padding to avoid cutted letters

Decrease text-label-position-toleranc and text-margin to allow more
content visible

kent1D commented:
Villeurbanne shouldn't be shown in place of Lyon:

<img width="1380" alt="villeurbanne_vs_lyon" src="https://user-images.githubusercontent.com/178744/42455228-f0222adc-8391-11e8-95af-342c7f340e1a.png">

Lorient versus Lanester:

<img width="1380" alt="lorient_vs_lanester" src="https://user-images.githubusercontent.com/178744/42455316-3a719e88-8392-11e8-82e2-ce764153e924.png">

Around Paris, Bobigny, Créteil, Cergy, Versailles, Evry, Nanterre et Melun should have priority against Saint-Denis, Montreuil, Boulogne-Billancourt and Argenteuil, even if they are smaller:

<img width="1380" alt="ile_de_france" src="https://user-images.githubusercontent.com/178744/42455683-38980ad8-8393-11e8-9df0-52afe05f99ea.png">

yohanboniface commented on forte.yml:
```diff
@@ -301,8 +301,9 @@ Layer:
-            geometry, '{lang}' as lang, type, capital, NULL as ldir,
+            geometry, '{lang}' as lang, type, NULL as ldir,
             CASE WHEN type='city' THEN 1 WHEN type='state' THEN 2 ELSE 100 END AS prio,
+            CASE WHEN capital='yes' THEN 1 WHEN capital IN ('4','5','6','7','8','9','10') THEN capital::INT ELSE 100 END AS capital,
```
There are a bunch of `capital=2` around, it seems, for islands. I think we should deal with that.

kent1D replied:
You are right, I looked only in my database (with all Europe) and there are 9 with `capital=2` and 37 with `capital=3`

I should use more the community tools, cf https://taginfo.openstreetmap.org/keys/capital#values

Should be ok with: https://github.com/tilery/pianoforte/pull/45/commits/e7f34b0a8612013095545f9a973b4fccbae121af

yohanboniface commented on forte.yml:
```diff
@@ -301,8 +301,9 @@ Layer:
         ( SELECT
-            geometry, '{lang}' as lang, type, capital, NULL as ldir,
+            geometry, '{lang}' as lang, type, NULL as ldir,
             CASE WHEN type='city' THEN 1 WHEN type='state' THEN 2 ELSE 100 END AS prio,
```
Are we sure we cant the `place=state` to be ordered even after cities with `capital=10` ?

kent1D replied:
Mmmh ok

https://github.com/tilery/pianoforte/pull/45/commits/14f4e0a473d82935829753fcc30bdd843f3d0b6d changes it like this:

* Capital = yes
* Capital IN 2,3,4 (regionals)
* State
* Other capitals

What do you think

yohanboniface commented:
> Villeurbanne shouldn't be shown in place of Lyon:

have you investigated why? Seems to me that Lyon has priority over Villeurbanne on every order column, even without this PR. Sounds like a bug (maybe the missing `NULLS LAST` ?).

kent1D committed:
There are cities or towns (mainly in islands, but not only) where the
"capital" tag value is 2 or 3.

For capital === 3 (37
times:https://taginfo.openstreetmap.org/tags/capital=3#overview)

* Nouméa 3 (New Caledonia): https://www.openstreetmap.org/node/291987644
* Tripoli 3 (Lebanon):
https://www.openstreetmap.org/node/258516078#map=18/34.44547/35.82349
* Saint-Louis (Senegal): http://www.openstreetmap.org/node/249167293

For capital === 2 (9 times:
https://taginfo.openstreetmap.org/tags/capital=2#overview)

kent1D committed:
Better order by prio

Capital = yes
Capital IN 2,3,4 (regionals)
State
Other capitals

kent1D committed:
More precise
